scripts/gen_lib.lua

```diff
@@ -0,0 +1,71 @@
+lpeg = require('lpeg')
```

Comment:
Well it is a library used by `scripts/` and unlike `dispatch_deprecated.lua` (which actually defines stuff exposed in the nvim binary, and thus belong in `src/`) it is not really api source. As you mention `test/` contains not only tests but also libraries for tests, is this not analogous?